great_expectations/validator/v1_validator.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import annotations
+
+from enum import Enum
+from typing import TYPE_CHECKING
+
+from great_expectations.core.expectation_validation_result import (
+    ExpectationSuiteValidationResult,
+    ExpectationValidationResult,
+)
+from great_expectations.validator.validator import calc_validation_statistics
+
+if TYPE_CHECKING:
+    from great_expectations.core import ExpectationSuite
+    from great_expectations.core.batch_config import BatchConfig
+    from great_expectations.data_context import AbstractDataContext
+    from great_expectations.expectations.expectation import (
+        Expectation,
+        ExpectationConfiguration,
+    )
+
+
+class ResultFormat(str, Enum):
+    BOOLEAN_ONLY = "BOOLEAN_ONLY"
+    BASIC = "BASIC"
+    SUMMARY = "SUMMARY"
+    COMPLETE = "COMPLETE"
+
+
+class Validator:
+    """Validator.
+
+    Responsible for running expectations on a batch configuration.
+    """
+
+    context: AbstractDataContext
+    batch_config: BatchConfig
+    batch_asset_options: dict | None
+    result_format: ResultFormat
+
+    def __init__(
+        self,
+        context: AbstractDataContext,
+        batch_config: BatchConfig,
+        result_format: ResultFormat = ResultFormat.SUMMARY,
+        batch_asset_options: dict | None = None,
+    ) -> None:
+        self.context = context
+        self.batch_config = batch_config
+        self.batch_asset_options = batch_asset_options
+        self.result_format = result_format
+
+    def validate_expectation(
+        self, expectation: Expectation
+    ) -> ExpectationValidationResult:
+        """Run a single expectation against the batch config"""
+        results = self._validate_expectation_configs([expectation.configuration])
+
+        assert len(results) == 1
```

Comment:
Could we raise a `ValueError` instead? We can bypass asserts with a Python flag

Reply:
I don't have super strong opinions here, but here was why I just put an `assert`: AFAIK we know that if we pass in a single configuration, we'll get out exactly one. In the past, I've used asserts when the point is to tell mypy or future devs something we know has to be true that mypy can't infer. Whereas a value error is about saying a bad value was passed into the function.

But I could totally not be following best practices. WDYT?

Reply:
Ah I didn't realize this was an invariant of sorts - yeah I agree with your decision here. I'd love to codify this distinction in our internal coding standards